test/extended/router/h2spec.go

```diff
@@ -108,30 +148,53 @@ var _ = g.Describe("[sig-network-edge][Conformance][Area:Networking][Feature:Rou
 			if iterations := lookupEnv("ROUTER_H2SPEC_SAMPLE", ""); iterations != "" {
 				n, err := strconv.Atoi(iterations)
 				o.Expect(err).NotTo(o.HaveOccurred())
-				runConformanceTestsAndLogAggregateFailures(oc, routeTypeTests, n)
+				o.Expect(n).To(o.BeNumerically(">", 0))
+				runConformanceTestsAndLogAggregateFailures(oc, host, "h2spec", n)
 				return
 			}
 
-			for _, t := range routeTypeTests {
-				g.By(fmt.Sprintf("[%s] Running h2spec conformance tests against %q", t.routeType, t.hostname))
-
-				testSuites := runConformanceTests(oc, t)
-				o.Expect(testSuites).ShouldNot(o.BeEmpty())
+			testSuites, err := runConformanceTests(oc, host, "h2spec")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(testSuites).ShouldNot(o.BeEmpty())
 
-				failures := failingTests(testSuites)
-				failureCount := len(failures)
+			failures := failingTests(testSuites)
+			failureCount := len(failures)
 
-				g.By("Analyzing results")
-				for _, f := range failures {
-					if _, exists := t.knownFailures[f.ID()]; exists {
-						failureCount -= 1
-						e2e.Logf("[%s] TestCase ID: %q is a known failure; ignoring", t.routeType, f.ID())
-					} else {
-						e2e.Logf("[%s] TestCase ID: %q (%q) ****FAILED****", t.routeType, f.ID(), f.TestCase.ClassName)
-					}
+			g.By("Analyzing results")
+			// https://github.com/haproxy/haproxy/issues/471#issuecomment-591420924
+			//
+			// 5. Streams and Multiplexing
+			//   5.4. Error Handling
+			//     5.4.1. Connection Error Handling
+			//       using source address 10.131.0.37:34742
+			//       × 2: Sends an invalid PING frame to receive GOAWAY frame
+			//         -> An endpoint that encounters a connection error SHOULD first send a GOAWAY frame
+			//            Expected: GOAWAY Frame (Error Code: PROTOCOL_ERROR)
+			//              Actual: Connection closed
+			//
+			// 6. Frame Definitions
+			//   6.9. WINDOW_UPDATE
+			//     6.9.1. The Flow-Control Window
+			//       using source address 10.131.0.37:34816
+			//       × 2: Sends multiple WINDOW_UPDATE frames increasing the flow control window to above 2^31-1
+			//         -> The endpoint MUST sends a GOAWAY frame with a FLOW_CONTROL_ERROR code.
+			//            Expected: GOAWAY Frame (Error Code: FLOW_CONTROL_ERROR)
+			//              Actual: Connection closed
+			//
+			// 147 tests, 145 passed, 0 skipped, 2 failed
+			knownFailures := map[string]bool{
+				"http2/5.4.1.2": true,
+				"http2/6.9.1.2": true,
+			}
+			for _, f := range failures {
+				if _, exists := knownFailures[f.ID()]; exists {
+					failureCount -= 1
```

Comment:
Why not do this section with gomega for consistency in the output?

Reply:
I.e., something like `o.Expect(knownFailures).ShouldNot(HaveKey(f.ID()))`

Reply:
The purpose is to gather all of them that fail and report them wholesale as opposed to failing on the first one that fails.